common/src/main/java/io/netty/util/concurrent/UnorderedThreadPoolEventExecutor.java

```diff
@@ -0,0 +1,256 @@
+ * Copyright 2016 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package io.netty.util.concurrent;
+
+import io.netty.util.internal.logging.InternalLogger;
+import io.netty.util.internal.logging.InternalLoggerFactory;
+
+import java.util.Collections;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Set;
+import java.util.concurrent.Callable;
+import java.util.concurrent.Delayed;
+import java.util.concurrent.RejectedExecutionHandler;
+import java.util.concurrent.RunnableScheduledFuture;
+import java.util.concurrent.ScheduledThreadPoolExecutor;
+import java.util.concurrent.ThreadFactory;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * {@link EventExecutor} implementation which makes no guarantees about the ordering of task execution that
+ * are submitted to it as it utilize the configured {@code corePoolSize}.
+ * This implementation is most useful for protocols that do not need strict ordering.
+ *
+ * <strong>Because it provides no ordering care should be taken when using it!</strong>
+ */
+public final class UnorderedThreadPoolEventExecutor extends ScheduledThreadPoolExecutor implements EventExecutor {
+    private static final InternalLogger logger = InternalLoggerFactory.getInstance(
+            UnorderedThreadPoolEventExecutor.class);
+
+    private final Promise<?> terminationFuture = GlobalEventExecutor.INSTANCE.newPromise();
+    private final Set<EventExecutor> executorSet = Collections.singleton((EventExecutor) this);
+
+    /**
+     * Calls {@link UnorderedThreadPoolEventExecutor#UnorderedThreadPoolEventExecutor(int, ThreadFactory)}
+     * using {@link DefaultThreadFactory}.
+     */
+    public UnorderedThreadPoolEventExecutor(int corePoolSize) {
+        this(corePoolSize, new DefaultThreadFactory(UnorderedThreadPoolEventExecutor.class));
+    }
+
+    /**
+     * See {@link ScheduledThreadPoolExecutor#ScheduledThreadPoolExecutor(int, ThreadFactory)}
+     */
+    public UnorderedThreadPoolEventExecutor(int corePoolSize, ThreadFactory threadFactory) {
+        super(corePoolSize, threadFactory);
+    }
+
+    /**
+     * Calls {@link UnorderedThreadPoolEventExecutor#UnorderedThreadPoolEventExecutor(int,
+     * ThreadFactory, java.util.concurrent.RejectedExecutionHandler)} using {@link DefaultThreadFactory}.
+     */
+    public UnorderedThreadPoolEventExecutor(int corePoolSize, java.util.concurrent.RejectedExecutionHandler handler) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Unnecessary use of fully qualified name 'java.util.concurrent.RejectedExecutionHandler' due to existing import 'java.util.concurrent.RejectedExecutionHandler' [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://garage.netty.io/sonarqube/coding_rules#rule_key=pmd%3AUnnecessaryFullyQualifiedName)